### ISSUE TITLE
[Automation API] Error message improvements

### ISSUE THREAD

**Issue by roderik** [pulumi/pulumi#6041]
I made a simple error, a duplicate name. The error output was

```
 stdout: Updating (clustertest)

View Live: https://app.pulumi.com/roderik/bpaastest/clustertest/updates/4


    pulumi:pulumi:Stack bpaastest-clustertest running 
    pulumi:pulumi:Stack bpaastest-clustertest running error: Unhandled exception: Error: Program run without the Pulumi engine available; re-run using the `pulumi` CLI
    bpaas:cluster v4-clustertest  
    bpaas:cluster v4-clustertest  error: Duplicate resource URN 'urn:pulumi:clustertest::bpaastest::bpaas:cluster::v4-clustertest'; try giving it a unique name
    pulumi:pulumi:Stack bpaastest-clustertest **failed** 1 error
    bpaas:cluster v4-clustertest **failed** 1 error
 
Diagnostics:
  pulumi:pulumi:Stack (bpaastest-clustertest):
    error: Unhandled exception: Error: Program run without the Pulumi engine available; re-run using the `pulumi` CLI
        at requireTestModeEnabled (/Users/roderik/Development/bpaas-launchpad/node_modules/@pulumi/pulumi/runtime/settings.js:124:15)
        at Object.getMonitor (/Users/roderik/Development/bpaas-launchpad/node_modules/@pulumi/pulumi/runtime/settings.js:206:13)
        at Object.registerResource (/Users/roderik/Development/bpaas-launchpad/node_modules/@pulumi/pulumi/runtime/resource.js:212:32)
        at new Resource (/Users/roderik/Development/bpaas-launchpad/node_modules/@pulumi/pulumi/resource.js:215:24)
        at new CustomResource (/Users/roderik/Development/bpaas-launchpad/node_modules/@pulumi/pulumi/resource.js:307:9)
        at new ProviderResource (/Users/roderik/Development/bpaas-launchpad/node_modules/@pulumi/pulumi/resource.js:336:9)
        at new Provider (/Users/roderik/Development/bpaas-launchpad/node_modules/@pulumi/provider.ts:51:9)
        at createCore (/Users/roderik/Development/bpaas-launchpad/packages/launchpad-api/node_modules/@pulumi/cluster.ts:536:25)
        at new Cluster (/Users/roderik/Development/bpaas-launchpad/packages/launchpad-api/node_modules/@pulumi/cluster.ts:1320:22)
        at ClusterResource.eksCluster (/Users/roderik/Development/bpaas-launchpad/packages/launchpad-api/src/entities/clusters/cluster.resource.ts:131:25)
 
  bpaas:cluster (v4-clustertest):
    error: Duplicate resource URN 'urn:pulumi:clustertest::bpaastest::bpaas:cluster::v4-clustertest'; try giving it a unique name
 
Resources:
    2 unchanged

Duration: 2s
```

## Expected Behavior

The error should just be

```
  bpaas:cluster (v4-clustertest):
    error: Duplicate resource URN 'urn:pulumi:clustertest::bpaastest::bpaas:cluster::v4-clustertest'; try giving it a unique name
 ```

And the first error should not happen, we are running via the automation API.

**Comment by roderik** [comment]
This run actually stalls at the end (cfr #5531) until I ctrl-c it, while it completed successfully. 

```
➜  launchpad-api (feat/bare-pulumi-setup) ✗ yarn ts-node ./tests/infra/cluster.pulumi.ts                                      
yarn run v1.22.10
$ /Users/roderik/Development/bpaas-launchpad/node_modules/.bin/ts-node ./tests/infra/cluster.pulumi.ts
Refreshing (clustertest)


View Live: https://app.pulumi.com/roderik/bpaastest/clustertest/updates/14




 

Resources:

Duration: 1s


Destroying (clustertest)


View Live: https://app.pulumi.com/roderik/bpaastest/clustertest/updates/15




 

Resources:

Duration: 1s


The resources in the stack have been deleted, but the history and configuration associated with the stack are still maintained. 
If you want to remove the stack completely, run 'pulumi stack rm clustertest'.

Updating (clustertest)


View Live: https://app.pulumi.com/roderik/bpaastest/clustertest/updates/16




 +  pulumi:pulumi:Stack bpaastest-clustertest creating 

 +  pulumi:pulumi:Stack bpaastest-clustertest creating error: Unhandled exception: Error: Program run without the Pulumi engine available; re-run using the `pulumi` CLI

 +  bpaas:cluster v4-clustertest2 creating 

 +  pulumi:providers:aws eu-west-3 creating 

 +  pulumi:providers:aws eu-west-3 created 

 +  eks:index:Cluster v4-clustertest2 creating 

 +  eks:index:ServiceRole v4-clustertest2-eksRole creating 

 +  aws:iam:Role v4-clustertest2-eksRole-role creating 

 +  aws:ec2:SecurityGroup v4-clustertest2-eksClusterSecurityGroup creating 

 +  aws:iam:Role v4-clustertest2-eksRole-role created 

 +  aws:iam:RolePolicyAttachment v4-clustertest2-eksRole-90eb1c99 creating 
 +  aws:iam:RolePolicyAttachment v4-clustertest2-eksRole-4b490823 creating 

 +  aws:ec2:SecurityGroup v4-clustertest2-eksClusterSecurityGroup created 

 +  aws:ec2:SecurityGroupRule v4-clustertest2-eksClusterInternetEgressRule creating 

 +  aws:iam:RolePolicyAttachment v4-clustertest2-eksRole-90eb1c99 created 

 +  aws:iam:RolePolicyAttachment v4-clustertest2-eksRole-4b490823 created 

 +  aws:eks:Cluster v4-clustertest2-eksCluster creating 

 +  aws:ec2:SecurityGroupRule v4-clustertest2-eksClusterInternetEgressRule created 

 +  aws:eks:Cluster v4-clustertest2-eksCluster created 

 +  pulumi:pulumi:Stack bpaastest-clustertest creating error: an unhandled error occurred: Program run without the Pulumi engine available; re-run using the `pulumi` CLI

 +  pulumi:pulumi:Stack bpaastest-clustertest **creating failed** 2 errors

 

Diagnostics:

  pulumi:pulumi:Stack (bpaastest-clustertest):
    error: Unhandled exception: Error: Program run without the Pulumi engine available; re-run using the `pulumi` CLI
        at requireTestModeEnabled (/Users/roderik/Development/bpaas-launchpad/node_modules/@pulumi/pulumi/runtime/settings.js:124:15)
        at Object.getMonitor (/Users/roderik/Development/bpaas-launchpad/node_modules/@pulumi/pulumi/runtime/settings.js:206:13)
        at Object.registerResource (/Users/roderik/Development/bpaas-launchpad/node_modules/@pulumi/pulumi/runtime/resource.js:212:32)
        at new Resource (/Users/roderik/Development/bpaas-launchpad/node_modules/@pulumi/pulumi/resource.js:215:24)
        at new CustomResource (/Users/roderik/Development/bpaas-launchpad/node_modules/@pulumi/pulumi/resource.js:307:9)
        at new ProviderResource (/Users/roderik/Development/bpaas-launchpad/node_modules/@pulumi/pulumi/resource.js:336:9)

        at new Provider (/Users/roderik/Development/bpaas-launchpad/node_modules/@pulumi/provider.ts:51:9)
        at createCore (/Users/roderik/Development/bpaas-launchpad/packages/launchpad-api/node_modules/@pulumi/cluster.ts:536:25)

        at new Cluster (/Users/roderik/Development/bpaas-launchpad/packages/launchpad-api/node_modules/@pulumi/cluster.ts:1320:22)
        at ClusterResource.eksCluster (/Users/roderik/Development/bpaas-launchpad/packages/launchpad-api/src/entities/clusters/cluster.resource.ts:129:25)
    error: an unhandled error occurred: Program run without the Pulumi engine available; re-run using the `pulumi` CLI
 

Resources:
    + 11 created

Duration: 12m5s


info: [runtime] Cluster is ready
CommandError: code: 255
 stdout: Updating (clustertest)

View Live: https://app.pulumi.com/roderik/bpaastest/clustertest/updates/16


 +  pulumi:pulumi:Stack bpaastest-clustertest creating 
 +  pulumi:pulumi:Stack bpaastest-clustertest creating error: Unhandled exception: Error: Program run without the Pulumi engine available; re-run using the `pulumi` CLI
 +  bpaas:cluster v4-clustertest2 creating 
 +  pulumi:providers:aws eu-west-3 creating 
 +  pulumi:providers:aws eu-west-3 created 
 +  eks:index:Cluster v4-clustertest2 creating 
 +  eks:index:ServiceRole v4-clustertest2-eksRole creating 
 +  aws:iam:Role v4-clustertest2-eksRole-role creating 
 +  aws:ec2:SecurityGroup v4-clustertest2-eksClusterSecurityGroup creating 
 +  aws:iam:Role v4-clustertest2-eksRole-role created 
 +  aws:iam:RolePolicyAttachment v4-clustertest2-eksRole-90eb1c99 creating 
 +  aws:iam:RolePolicyAttachment v4-clustertest2-eksRole-4b490823 creating 
 +  aws:ec2:SecurityGroup v4-clustertest2-eksClusterSecurityGroup created 
 +  aws:ec2:SecurityGroupRule v4-clustertest2-eksClusterInternetEgressRule creating 
 +  aws:iam:RolePolicyAttachment v4-clustertest2-eksRole-90eb1c99 created 
 +  aws:iam:RolePolicyAttachment v4-clustertest2-eksRole-4b490823 created 
 +  aws:eks:Cluster v4-clustertest2-eksCluster creating 
 +  aws:ec2:SecurityGroupRule v4-clustertest2-eksClusterInternetEgressRule created 
 +  aws:eks:Cluster v4-clustertest2-eksCluster created 
 +  pulumi:pulumi:Stack bpaastest-clustertest creating error: an unhandled error occurred: Program run without the Pulumi engine available; re-run using the `pulumi` CLI
 +  pulumi:pulumi:Stack bpaastest-clustertest **creating failed** 2 errors
 
Diagnostics:
  pulumi:pulumi:Stack (bpaastest-clustertest):
    error: Unhandled exception: Error: Program run without the Pulumi engine available; re-run using the `pulumi` CLI
        at requireTestModeEnabled (/Users/roderik/Development/bpaas-launchpad/node_modules/@pulumi/pulumi/runtime/settings.js:124:15)
        at Object.getMonitor (/Users/roderik/Development/bpaas-launchpad/node_modules/@pulumi/pulumi/runtime/settings.js:206:13)
        at Object.registerResource (/Users/roderik/Development/bpaas-launchpad/node_modules/@pulumi/pulumi/runtime/resource.js:212:32)
        at new Resource (/Users/roderik/Development/bpaas-launchpad/node_modules/@pulumi/pulumi/resource.js:215:24)
        at new CustomResource (/Users/roderik/Development/bpaas-launchpad/node_modules/@pulumi/pulumi/resource.js:307:9)
        at new ProviderResource (/Users/roderik/Development/bpaas-launchpad/node_modules/@pulumi/pulumi/resource.js:336:9)
        at new Provider (/Users/roderik/Development/bpaas-launchpad/node_modules/@pulumi/provider.ts:51:9)
        at createCore (/Users/roderik/Development/bpaas-launchpad/packages/launchpad-api/node_modules/@pulumi/cluster.ts:536:25)
        at new Cluster (/Users/roderik/Development/bpaas-launchpad/packages/launchpad-api/node_modules/@pulumi/cluster.ts:1320:22)
        at ClusterResource.eksCluster (/Users/roderik/Development/bpaas-launchpad/packages/launchpad-api/src/entities/clusters/cluster.resource.ts:129:25)
    error: an unhandled error occurred: Program run without the Pulumi engine available; re-run using the `pulumi` CLI
 
Resources:
    + 11 created

Duration: 12m5s


 stderr: 
 err?: 

    at Object.createCommandError (/Users/roderik/Development/bpaas-launchpad/packages/launchpad-api/node_modules/@pulumi/pulumi/x/automation/errors.js:71:17)
    at ChildProcess.<anonymous> (/Users/roderik/Development/bpaas-launchpad/packages/launchpad-api/node_modules/@pulumi/pulumi/x/automation/cmd.js:63:40)
    at ChildProcess.emit (node:events:376:20)
    at ChildProcess.EventEmitter.emit (node:domain:470:12)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:284:12) {
  commandResult: CommandResult {
    stdout: 'Updating (clustertest)\n' +
      '\n' +
      'View Live: https://app.pulumi.com/roderik/bpaastest/clustertest/updates/16\n' +
      '\n' +
      '\n' +
      ' +  pulumi:pulumi:Stack bpaastest-clustertest creating \n' +
      ' +  pulumi:pulumi:Stack bpaastest-clustertest creating error: Unhandled exception: Error: Program run without the Pulumi engine available; re-run using the `pulumi` CLI\n' +
      ' +  bpaas:cluster v4-clustertest2 creating \n' +
      ' +  pulumi:providers:aws eu-west-3 creating \n' +
      ' +  pulumi:providers:aws eu-west-3 created \n' +
      ' +  eks:index:Cluster v4-clustertest2 creating \n' +
      ' +  eks:index:ServiceRole v4-clustertest2-eksRole creating \n' +
      ' +  aws:iam:Role v4-clustertest2-eksRole-role creating \n' +
      ' +  aws:ec2:SecurityGroup v4-clustertest2-eksClusterSecurityGroup creating \n' +
      ' +  aws:iam:Role v4-clustertest2-eksRole-role created \n' +
      ' +  aws:iam:RolePolicyAttachment v4-clustertest2-eksRole-90eb1c99 creating \n' +
      ' +  aws:iam:RolePolicyAttachment v4-clustertest2-eksRole-4b490823 creating \n' +
      ' +  aws:ec2:SecurityGroup v4-clustertest2-eksClusterSecurityGroup created \n' +
      ' +  aws:ec2:SecurityGroupRule v4-clustertest2-eksClusterInternetEgressRule creating \n' +
      ' +  aws:iam:RolePolicyAttachment v4-clustertest2-eksRole-90eb1c99 created \n' +
      ' +  aws:iam:RolePolicyAttachment v4-clustertest2-eksRole-4b490823 created \n' +
      ' +  aws:eks:Cluster v4-clustertest2-eksCluster creating \n' +
      ' +  aws:ec2:SecurityGroupRule v4-clustertest2-eksClusterInternetEgressRule created \n' +
      ' +  aws:eks:Cluster v4-clustertest2-eksCluster created \n' +
      ' +  pulumi:pulumi:Stack bpaastest-clustertest creating error: an unhandled error occurred: Program run without the Pulumi engine available; re-run using the `pulumi` CLI\n' +
      ' +  pulumi:pulumi:Stack bpaastest-clustertest **creating failed** 2 errors\n' +
      ' \n' +
      'Diagnostics:\n' +
      '  pulumi:pulumi:Stack (bpaastest-clustertest):\n' +
      '    error: Unhandled exception: Error: Program run without the Pulumi engine available; re-run using the `pulumi` CLI\n' +
      '        at requireTestModeEnabled (/Users/roderik/Development/bpaas-launchpad/node_modules/@pulumi/pulumi/runtime/settings.js:124:15)\n' +
      '        at Object.getMonitor (/Users/roderik/Development/bpaas-launchpad/node_modules/@pulumi/pulumi/runtime/settings.js:206:13)\n' +
      '        at Object.registerResource (/Users/roderik/Development/bpaas-launchpad/node_modules/@pulumi/pulumi/runtime/resource.js:212:32)\n' +
      '        at new Resource (/Users/roderik/Development/bpaas-launchpad/node_modules/@pulumi/pulumi/resource.js:215:24)\n' +
      '        at new CustomResource (/Users/roderik/Development/bpaas-launchpad/node_modules/@pulumi/pulumi/resource.js:307:9)\n' +
      '        at new ProviderResource (/Users/roderik/Development/bpaas-launchpad/node_modules/@pulumi/pulumi/resource.js:336:9)\n' +
      '        at new Provider (/Users/roderik/Development/bpaas-launchpad/node_modules/@pulumi/provider.ts:51:9)\n' +
      '        at createCore (/Users/roderik/Development/bpaas-launchpad/packages/launchpad-api/node_modules/@pulumi/cluster.ts:536:25)\n' +
      '        at new Cluster (/Users/roderik/Development/bpaas-launchpad/packages/launchpad-api/node_modules/@pulumi/cluster.ts:1320:22)\n' +
      '        at ClusterResource.eksCluster (/Users/roderik/Development/bpaas-launchpad/packages/launchpad-api/src/entities/clusters/cluster.resource.ts:129:25)\n' +
      '    error: an unhandled error occurred: Program run without the Pulumi engine available; re-run using the `pulumi` CLI\n' +
      ' \n' +
      'Resources:\n' +
      '    + 11 created\n' +
      '\n' +
      'Duration: 12m5s\n' +
      '\n',
    stderr: '',
    code: 255,
    err: undefined
  }
}
```

I thought it was due to me importing the CustomResource at the top like usual TS code, but switching to importing like you do in the unit test docs also does not work:

For reference the pulumi stack code

```
  const stack = await LocalWorkspace.createOrSelectStack({
    stackName: 'clustertest',
    projectName: 'bpaastest',
    program: async () => {
      const { ClusterResource } = await import('../../src/entities/clusters/cluster.resource');
      // const gke = new ClusterResource(`v4-clustertest1`, ClusterProvider.GKE, ClusterRegion.EUROPE, ClusterSize.SMALL);
      const eks = new ClusterResource(`v4-clustertest2`, ClusterProvider.EKS, ClusterRegion.EUROPE, ClusterSize.SMALL);
      // const aks = new ClusterResource(`v4-clustertest3`, ClusterProvider.AKS, ClusterRegion.EUROPE, ClusterSize.SMALL);
      return {
        // kubeconfigGKE: gke.kubeconfig,
        // kubeconfigAKS: aks.kubeconfig,
        kubeconfigEKS: eks.kubeconfig,
      };
    },
  });
```

And the custom resource: https://gist.github.com/roderik/4bf4b2b6a94ea6a2cb506388dca2c096

**Comment by roderik** [comment]
This solely happens with the @pulumi/eks package. it might not be automation API enabled?

**Comment by roderik** [comment]
There is more going on, i keep getting hit with this error even for GKE clusters, https://gist.github.com/roderik/ada6b6495ad4b464a79bfed2848887a4

FYI: I opened support ticket #402 for some one on one support

**Comment by viveklak** [comment]
@roderik if possible, a simplified repro would be quite useful here. i.e. a program which works with pulumi CLI but doesn't with automation api. Can't think of anything obvious as to why the environment variable storing the monitor address would be omitted on automation api (prompting the error you are seeing).

**Comment by charlex** [comment]
I'm having this happen as well.

**Comment by charlex** [comment]
@viveklak I think @EvanBoyle has created a repro here: https://github.com/pulumi/pulumi/issues/6057

**Comment by viveklak** [comment]
@charlex thanks. I was going to reach out to @EvanBoyle! 

@EvanBoyle should this be closed duplicate of #6057 since you seem to be tracking things there?

**Comment by EvanBoyle** [comment]
There's now a linked PR that will close both issues out.
